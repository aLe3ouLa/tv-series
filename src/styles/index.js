import React from "react";

import { Reset, Typography } from "./defaults";
import { color } from "./variables/colors";
import { spacing } from "./variables/spacings";
import { breakpoint } from "./variables/breakpoints";

export const Defaults = () => (
  <>
    <Reset />
    <Typography />
  </>
);

export const theme = {
  color,
  spacing,
  breakpoint,
};
