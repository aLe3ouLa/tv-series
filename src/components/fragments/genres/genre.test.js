import React from "react";
import Genres from "./genre";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../styles";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Genre component", () => {
  afterEach(cleanup);

  const mockprops = {
    genres: ["test"],
  };


  it("should render genre component without crashing", () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Genres {...mockprops} />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(getByTestId("genre-list").querySelectorAll("span").length).toBe(1);
  });
});
