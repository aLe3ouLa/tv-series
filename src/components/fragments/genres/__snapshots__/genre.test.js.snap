// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre component should render genre component without crashing 1`] = `
<div>
  <ul
    class="sc-AxjAm byYaqZ"
  >
    <li
      class="sc-AxirZ vMuzg"
      data-testid="genre-list"
    >
      <span
        class="genre-label"
      >
        test
      </span>
    </li>
  </ul>
</div>
`;
